src/main/java/dev/morling/onebrc/CalculateAverage_artsiomkorzun.java

```diff
@@ -218,11 +227,10 @@ public static void update(long address, int value) {
             UNSAFE.putInt(address + 16, cnt);
 
             if (value < min) {
-                UNSAFE.putShort(address + 20, (short) value);
+                UNSAFE.putShort(address + 20, value);
             }
-
-            if (value > max) {
-                UNSAFE.putShort(address + 22, (short) value);
+            else if (value > max) {
```

Comment:
Nice, I guess it did make a difference? You could well be breaking 2 seconds now!

Reply:
A very small difference, which I can't confirm with 100% certainty. Not filling memory with zeros gives around 10ms as well. Lets see how it works out on the eval machine.